crates/ol/msg-types/src/withdrawal.rs

```diff
@@ -0,0 +1,332 @@
+//! Withdrawal message types for bridge gateway account communication.
+
+use strata_acct_types::VarVec;
+use strata_codec::{Codec, CodecError, Decoder, Encoder};
+use strata_identifiers::SubjectId;
+
+/// Message type ID for withdrawal initiation.
+pub const WITHDRAWAL_MSG_TYPE_ID: u16 = 0x03;
+
+/// Message type ID for withdrawal fee bump.
+pub const WITHDRAWAL_FEE_BUMP_MSG_TYPE_ID: u16 = 0x04;
+
+/// Message type ID for withdrawal rejection.
+pub const WITHDRAWAL_REJECTION_MSG_TYPE_ID: u16 = 0x05;
+
+/// Maximum length for withdrawal destination descriptor.
+pub const MAX_WITHDRAWAL_DESC_LEN: usize = 255;
+
+/// Message data for withdrawal initiation to the bridge gateway account.
+///
+/// This message type is sent by accounts that want to trigger a withdrawal.
+/// The value sent with the message should be equal to the predetermined
+/// static withdrawal size.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct WithdrawalMsgData {
+    /// Fees in satoshis to be paid to the operator.
+    pub fees: u32,
+    /// Bitcoin Output Script Descriptor describing the withdrawal output.
+    pub dest_desc: VarVec<u8>,
+}
+
+impl WithdrawalMsgData {
+    /// Create a new withdrawal message data instance.
+    pub fn new(fees: u32, dest_desc: Vec<u8>) -> Option<Self> {
+        // Ensure the destination descriptor isn't too long
```

Comment:
this is for the bitcoin address, right? Some bitcoin addresses, like bech32, have some error correction associated I think, so we may want to add this at some point.